src/compiler/compiler.c

```
@@ -131,6 +137,52 @@ IR_CTX* makeContext(AST_NODE* tree) {
  */
-void compile(IR_CTX* ctx, char* outputFileName) {
+void compile(IR_CTX* ctx, FILE* out) {
+
+	uint8_t* buff = malloc(sizeof(uint8_t) * 512);
+	int i = 0;
+
+	int h = hashstr("main");
+
+	if(hashGet(ctx->nodeMap, h) == NULL) {
+		printf("Error: the main function wasn't defined!\n");
+		return;
```

cppcheck (2.10-2)

[error] Memory leak

(memleak)